src/prefect/orion/schemas/core.py

```diff
@@ -314,11 +318,30 @@ def populate_deprecated_fields(cls, values):
         """
         if not values.get("retries", None) and values.get("max_retries", 0) != 0:
             values["retries"] = values["max_retries"]
+
         if (
             not values.get("retry_delay", None)
             and values.get("retry_delay_seconds", 0) != 0
         ):
             values["retry_delay"] = values["retry_delay_seconds"]
+
+        return values
+
+    @root_validator
+    def validate_retry_and_jitter_inputs(cls, values):
+        """
+        Limits the number of configured retry delays to a maximum of 50. And ensures
+        that the relative jitter factor is positive.
+        """
+        if isinstance(values["retry_delay"], list) and (
+            len(values["retry_delay"]) > 50
+        ):
+            values["retry_delay"] = values["retry_delay"][:50]
+
+        configured_relative_jitter = values.get("retry_jitter_factor", None)
+        if configured_relative_jitter is not None and configured_relative_jitter < 0:
+            values["retry_jitter_factor"] = None
```

Comment:
Is this enforcing that it should be greater than zero? We should also have this raise an error.

Reply:
this isn't doing the enforcing, but I figured we might want to fully orchestrate otherwise valid task runs. If we remove the check the actual guard in `clamped_poisson_interval` will raise a `ValueError`